src/arroyo_reduction/redis_model_store.py

```diff
@@ -206,3 +206,54 @@ def listener_thread():
         thread.start()
         logger.info("Started model update listener thread")
 
+    def get_model_loading_state(self):
+        """
+        Get the current model loading state from Redis
+        
+        Returns:
+            dict: Dictionary with is_loading_model (bool) and loading_model_type (str or None)
+        """
+        if self.redis_client is None:
+            logger.warning("Redis client not available")
+            return {"is_loading_model": False, "loading_model_type": None}
+        
+        try:
+            # Get values from Redis
+            is_loading = self.redis_client.get("model_loading_state")
+            loading_type = self.redis_client.get("loading_model_type")
+            
+            # Process is_loading value carefully
+            if is_loading is not None:
```

Comment:
The the logic to check for not null and seriolize from binary to string is copy/pasted. Let's put that into a generic method.

```
def get_value(redis_key)
    raw_value = self.redis_client.get(redis_key)
    if raw_value is not None:
        if isinstance(raw_value, bytes):
            value_str = loading_type.decode('utf-8')
        else:
            value_str = str(raw_value)
            
        if not value_str:  # Empty string
            value_str = None
    else:
        value_str = None
    return value_str

```